### PR TITLE
new pipe sydems

### PR DESCRIPTION
**Pull request by screret**
FIXES:
- fixes fluid pipes being P2P
- adds pipe restrictors
- adds gt6StylePipesCables config
- reworks how pipenet walkers work
- removes usage of `Direction#values()`

### CODE REVIEW

File: src/main/java/com/gregtechceu/gtceu/common/cover/FacadeCover.java [hunk 1]
```diff
@@ -57,7 +57,7 @@ public boolean shouldRenderPlate() {
     /**
      * @return If the pipe this is placed on and a pipe on the other side should be able to connect
      */
-    public boolean blockPipePassThrough() {
+    public boolean canPipePassThrough() {
         return false;
```
Comment (mikerooni) [inline src/main/java/com/gregtechceu/gtceu/common/cover/FacadeCover.java:61]: Not sure if the returned value also needs to change, as the renamed method has different semantics here.

File: src/main/java/com/gregtechceu/gtceu/utils/EntityDamageUtil.java [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+     * @param entity the entity to damage
+     * @param damage the damage to apply
+     */
+    public static void applyHeatDamage(@NotNull LivingEntity entity, int damage) {
+        // do not attempt to damage by 0
+        if (damage <= 0) return;
+        if (!entity.isAlive()) return;
+        // fire/lava mobs cannot be burned
+        if (entity instanceof Blaze || entity instanceof MagmaCube ||
+                entity instanceof WitherSkeleton || entity instanceof WitherBoss)
```
Comment (mikerooni) [inline src/main/java/com/gregtechceu/gtceu/utils/EntityDamageUtil.java]: Would probably be better for compatibility to make this a tag (if entities can have tags - I'm not sure about that)

File: src/main/java/com/gregtechceu/gtceu/utils/EntityDamageUtil.java [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+     * @param entity the entity to damage
+     * @param damage the damage to apply
+     */
+    public static void applyFrostDamage(@NotNull LivingEntity entity, int damage) {
+        // do not attempt to damage by 0
+        if (damage <= 0) return;
+        if (!entity.isAlive()) return;
+        // snow/frost mobs cannot be chilled
+        if (entity instanceof SnowGolem || entity instanceof PolarBear || entity instanceof Stray)
+            return;
```
Comment (mikerooni) [inline src/main/java/com/gregtechceu/gtceu/utils/EntityDamageUtil.java]: Here as well

File: src/main/java/com/gregtechceu/gtceu/utils/EntityDamageUtil.java [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+    /**
+     * @param entity the entity to damage
+     * @param damage the damage to apply
+     */
+    public static void applyChemicalDamage(@NotNull LivingEntity entity, int damage) {
+        // do not attempt to damage by 0
+        if (damage <= 0) return;
+        if (!entity.isAlive()) return;
+        // skeletons cannot breathe in the toxins
+        if (entity instanceof AbstractSkeleton) return;
```
Comment (mikerooni) [inline src/main/java/com/gregtechceu/gtceu/utils/EntityDamageUtil.java]: Here as well

File: src/main/java/com/gregtechceu/gtceu/common/cover/FacadeCover.java (hunk 1 above)
Comment (screret) [reply src/main/java/com/gregtechceu/gtceu/common/cover/FacadeCover.java:61]: Nope, i also changed how it's used 
Also look at the javadoc, it defines the new semantics

File: src/main/java/com/gregtechceu/gtceu/utils/EntityDamageUtil.java [hunk 5]
```diff
@@ -72,18 +72,15 @@ public static void applyFrostDamage(@NotNull LivingEntity entity, int damage) {
...
-                CompoundTag compound = (CompoundTag) base;
-                if (compound.getShort("id") == FROST_WALKER_ID) {
-                    stack.hurtAndBreak(1, entity, ent -> ent.broadcastBreakEvent(EquipmentSlot.FEET));
-                    return;
-                }
+            if (stack.getEnchantmentLevel(Enchantments.FROST_WALKER) > 0) {
+                stack.hurtAndBreak(1, entity, ent -> ent.broadcastBreakEvent(EquipmentSlot.FEET));
+                return;
             }
         }
```
Comment (mikerooni) [inline src/main/java/com/gregtechceu/gtceu/utils/EntityDamageUtil.java]: Can't add a comment later but you forgor to use the chemical immune tag :trollface:

**Review by mikerooni — CHANGES_REQUESTED** [review]

File: src/main/java/com/gregtechceu/gtceu/utils/EntityDamageUtil.java [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+    /**
+     * @param entity the entity to damage
+     * @param damage the damage to apply
+     */
+    public static void applyChemicalDamage(@NotNull LivingEntity entity, int damage) {
+        // do not attempt to damage by 0
+        if (damage <= 0) return;
+        if (!entity.isAlive()) return;
+        // skeletons cannot breathe in the toxins
+        if (entity instanceof AbstractSkeleton) return;
```
Comment (mikerooni) [inline src/main/java/com/gregtechceu/gtceu/utils/EntityDamageUtil.java]:
```suggestion
        if (entity.getType().is(CustomTags.CHEMICAL_IMMUNE))
            return;
```

**Review by mikerooni — CHANGES_REQUESTED** [review]

**Review by mikerooni — APPROVED** [review]

**Review by mikerooni — APPROVED** [review]

**Pull request merged**